ui_framework/src/components/form/text_input.test.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import { render, shallow } from 'enzyme';
+import { requiredProps } from '../../test/required_props';
+import sinon from 'sinon';
+
+import {
+  KuiTextInput,
+  TEXTINPUT_SIZE
+} from './text_input';
+
+describe('KuiTextInput', () => {
+  test('renders', () => {
+    const component = (
+      <KuiTextInput
+        value="text input"
+        onChange={()=>{}}
+        {...requiredProps}
+      />
+    );
+
+    expect(render(component)).toMatchSnapshot();
+  });
+
+  describe('Props', () => {
+    test('placeholder', () => {
+      const component = (
+        <KuiTextInput
+          placeholder="placeholder"
+          onChange={()=>{}}
+        />
+      );
+
+      expect(render(component)).toMatchSnapshot();
+    });
+
+    test('value', () => {
+      const component = (
+        <KuiTextInput
+          value="value"
+          onChange={()=>{}}
+        />
+      );
+
+      expect(render(component)).toMatchSnapshot();
+    });
+
+    describe('autoFocus', () => {
```

Comment:
Can we update this test to assert that the focus is applied instead of checking a snapshot? We'll need to run `node scripts/jest -u` to remove the obsolete snapshots after making this change.

```jsx
import { render, shallow, mount } from 'enzyme';

    describe('autoFocus', () => {
      test('sets focus on the element', () => {
        const component = mount(
          <KuiTextInput
            autoFocus
            onChange={()=>{}}
            data-test-subj="input"
          />
        );

        expect(
          component.find('[data-test-subj="input"]').matchesElement(document.activeElement)
        ).toBe(true);
      });

      test('does not focus the element by default', () => {
        const component = mount(
          <KuiTextInput
            onChange={()=>{}}
            data-test-subj="input"
          />
        );

        expect(
          component.find('[data-test-subj="input"]').matchesElement(document.activeElement)
        ).toBe(false);
      });
    });
```

Reply:
Cool idea checking the autofocus behavior as behavior. Thanks! Much better tests.